packaging/pep517_backend/_backend.py

```diff
@@ -93,21 +114,29 @@ def build_sdist(  # noqa: WPS210, WPS430
     build_manpages_requested = BUILD_MANPAGES_CONFIG_SETTING in (
         config_settings or {}
     )
-    if build_manpages_requested:
-        Path('docs/man/man1/').mkdir(exist_ok=True, parents=True)
-        version_number = _get_package_distribution_version()
-        for rst_in in _generate_rst_in_templates():
-            _convert_rst_in_template_to_manpage(
-                rst_doc_template=rst_in.read_text(),
-                destination_path=rst_in.with_suffix('').with_suffix(''),
-                version_number=version_number,
-            )
-            rst_in.unlink()
-
-    return _setuptools_build_sdist(
-        sdist_directory=sdist_directory,
-        config_settings=config_settings,
-    )
+    original_src_dir = Path.cwd().resolve()
+    with _run_in_temporary_directory() as tmp_dir:
```

Comment:
It looks like `TemporaryDirectory` can be used here instead, since `os.chdir` is being called below. That means `_run_in_temporary_directory` and `_chdir_cm` can be removed, greatly simplifying this commit.

Reply:
@mattclay not really, the `TemporaryDirectory` CM does not `chdir` back to the original dir. So with a bare `chdir` below, we'd be influencing external calling environment (PEP 517 build front-end). We have to `chdir` back to prevent unintentional side effects, which is what said helper does.

Reply:
Ah, good point.